### ISSUE TITLE
[Favicons refactor] - "Amazon.com"'s favicon doesn't load in share menu

### ISSUE THREAD

**Issue by DianaABarladeanu** [mozilla-mobile/firefox-ios#22084]
**Steps to reproduce**
1. Install FF.
2. Open share menu on amazon.com (either with hamburger menu or with context menu).
3. Observe the favicon.

**Expected behavior**
Tha favicon is instantly and correctly loaded.

**Actual behavior**
The favicon doesn't load, a compass is displayed instead.

**Device & build information**
* Device: iPhone 15 (17.5)
* Build version: v131 (45403)
* First seen version: ? 

Notes
Attachments: 

https://github.com/user-attachments/assets/63cb8619-9887-4904-aed6-1d27fa652302

**Comment by OrlaM** [comment]
@ih-codes this one is a little different. The Amazon logo should exist in the cache at this point when the share sheet is open.

**Comment by ih-codes** [comment]
> @ih-codes this one is a little different. The Amazon logo should exist in the cache at this point when the share sheet is open.

Agreed, this one needs some investigation. I think I saw a few tickets where this share sheet doesn't seem to load favicons.
